pkg/prometheus/promcfg.go

```diff
@@ -320,6 +321,7 @@ func (cg *ConfigGenerator) Generate(
 	additionalAlertRelabelConfigs []byte,
 	additionalAlertManagerConfigs []byte,
 	ruleConfigMapNames []string,
+	isSupportEndpointSliceResource bool,
```

Comment:
A better variable name could be either `endpointSliceSupported` or `isEndpointSliceSupported`, both here and in `operator.go`.